sdk/identity/Azure.Identity/tests/ManagedIdentityCredentialSFLiveTests.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading.Tasks;
+using Azure.Core.TestFramework;
+using Azure.Security.KeyVault.Secrets;
+using NUnit.Framework;
+
+namespace Azure.Identity.Tests
+{
+    public class ManagedIdentityCredentialSFLiveTests : ManagedIdentityCredentialLiveTestBase
+    {
+        public ManagedIdentityCredentialSFLiveTests(bool isAsync) : base(isAsync)
+        {
+        }
+
+        [NonParallelizable]
+        [Test]
+        public async Task ValidateSystemAssignedIdentity()
+        {
+            if (string.IsNullOrEmpty(TestEnvironment.SFEnable) || !string.IsNullOrEmpty(TestEnvironment.UserAssignedVault))
+            {
+                Assert.Ignore();
+            }
+
+            using (ReadOrRestoreManagedIdentityEnvironment())
+            {
+                var vaultUri = new Uri(TestEnvironment.SystemAssignedVault);
+
+                CredentialPipeline pipeline = CredentialPipeline.GetInstance(InstrumentClientOptions(new TokenCredentialOptions { Transport = ServiceFabricManagedIdentitySource.GetServiceFabricMITransport() }));
+
+                var cred = new ManagedIdentityCredential(new ManagedIdentityClient(new ManagedIdentityClientOptions { Pipeline = pipeline, PreserveTransport = (Mode == RecordedTestMode.Playback) }));
+
+                // Hard code service version or recorded tests will fail: https://github.com/Azure/azure-sdk-for-net/issues/10432
+                var kvoptions = InstrumentClientOptions(new SecretClientOptions(SecretClientOptions.ServiceVersion.V7_0));
+
+                var kvclient = new SecretClient(vaultUri, cred, kvoptions);
```

Comment:
Do we really need to involve the SecretClient in these tests?

Reply:
Unfortunately yes to ensure we've gotten a token for the right identity. Some managed identity endpoints will unfortunately ignore the specified user assigned client id and give you back a token for the system assigned identity (which sometimes isn't event valid). So we use the Key Vault service as our "token validator".